### PR TITLE
Add space indentation handling

### LINKED ISSUE #2: Does not work with spaced indentation formatting
Only works with tabs

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
Fixes #2.

I divided the PR into 4 commits to make it easy to take part/s of it in case you don't like it as a whole. I can squash the commits into one if they're too granular for you.
## Changes:
- Modified extension.ts to handle spaces based indention (1edf329) 
- Simplified the working loop of block wrapping by moving handling of last and first row outside the loop (c2f657c)
- Removed some of the comments and trailing whitespaces (c2f657c)
- Added end to end tests for spaces/tabs block/line (a68c2f8)
- Added dependency on chai and fs-extra in tests (a68c2f8)
- Added typings usage - config file, updated npm scripts (a68c2f8)

### CODE REVIEW

**Comment by gdziadkiewicz** [comment]
@bgashler1 Will you be able to review it?

**Comment by bgashler1** [comment]
@gdziadkiewicz thank you for making this pull request!  Yes, I'll review it as soon as I'm able to.  I appreciate you giving back these improvements to the community.

**Comment by gdziadkiewicz** [comment]
@bgashler1 I just added settings for TravisCI. I tested it on my account -> [link to results](https://travis-ci.org/gdziadkiewicz/vscode-htmltagwrap/builds/148853179)  . Please remember to activate the project on your account and to add the badge to readme file after merge.

**Comment by bgashler1** [comment]
@gdziadkiewicz sorry for the delay in merging this in.  I've been very busy, but I finally had the opportunity to look it over it.  It looks good.  I just need to test and then I'll push an update out.  Thanks for contributing to this extension 💯

**Comment by gdziadkiewicz** [comment]
@bgashler1 No problem :) I've rerun the build on Travis to make sure that all tests still pass.

File: package.json [hunk 1]
```diff
@@ -30,11 +30,16 @@
...
         "compile": "node ./node_modules/vscode/bin/compile -watch -p ./",
-		"postinstall": "node ./node_modules/vscode/bin/install"
+		"postinstall": "typings install && node ./node_modules/vscode/bin/install",
+		"typings": "typings",
+		"test:LinuxAndOSX": "node ./node_modules/vscode/bin/test"
 	},
 	"devDependencies": {
-		"vscode": "0.11.x",
-		"typescript": "^1.8.5"
+		"chai": "*",
```
Comment (Tyriar) [inline package.json:38]: Shouldn't these deps be rooted at a minor version to keep the build stable?

File: test/extension.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import {copySync, CopyOptions, emptyDir} from 'fs-extra';
+import * as extension from '../src/extension';
+
+extension.activate();
+let extensionID = 'bradgashler.htmltagwrap';
+let samplesFolder = extensions.getExtension(extensionID).extensionPath + '/test/sampleFiles/';
+let tempFolder = samplesFolder + 'temp/';
+
+function parametrizedTest(startFilePath: string, expectedResultFilePath: string, selectionStart: Position, selectionEnd: Position, failMessage: string) {
+		let result: string;
```
Comment (Tyriar) [inline test/extension.test.ts:17]: Can u fix up indentation here?

**Comment by bgashler1** [comment]
Thanks @gdziadkiewicz for your contribution!  Merged in.

**Pull request merged**